third_party/proton/dialect/lib/ProtonGPUToLLVM/AllocateProtonGlobalScratchBuffer.cpp

```diff
@@ -24,8 +24,21 @@ struct AllocateProtonGlobalScratchBuffer
     MLIRContext *ctx = &getContext();
     OpBuilder builder(ctx);
 
-    assert(llvm::range_size(mod.getOps<FunctionOpInterface>()) == 1);
-    auto func = *mod.getOps<FunctionOpInterface>().begin();
+    llvm::errs() << "\n";
+    int numFuncOps = 0;
+    FunctionOpInterface func;
+    mod.walk([&](FunctionOpInterface op) {
```

Comment:
I guess we can just get the root function?

Reply:
I'm not sure. Maybe for now ignore intrinsics and revisit when we go to support device functions?

Reply:
OK